simplex_transforms/stan/transforms/Exponential/Exponential_functions.stan

```diff
@@ -0,0 +1,21 @@
+vector exponential_simplex_constrain_lp(vector y) {
+  int N = rows(y);
+  vector[N] x_pos = y - log1p_exp(-y);
+  real r = log_sum_exp(x_pos);
+  vector[N] x = exp(x_pos - r);
+  target += x_pos;
+  target += std_normal_lpdf(r - log(N));
+  target += sum(y) - N * r;
+  return x;
```

Comment:
I can just return `exp(x_pos - r)` instead of allocating to `x` to save on memory.

Reply:
Either way `x` would be allocated, I think, since that's needed in the model. Besides, our other implementations explicitly define `x`, so I prefer keeping that for consistency.